shared/AppInsightsCore/src/JavaScriptSDK/HelperFuncs.ts

```diff
@@ -116,9 +181,17 @@ export function detachEvent(obj: any, eventNameWithoutOn: string, handlerRef: an
  */
 export function normalizeJsName(name: string): string {
     let value = name;
-    let match = /([^\w\d_$])/g;
-    if (match.test(name)) {
-        value = name.replace(match, "_");
+
+    if (value && isString(value)) {
```

Comment:
Updated this as in a future PR I'm using this a lot more and the original version didn't quite matching the JavaScript specification.

Like a variable must not "start" with a digit, so while "0" is value "0a" is not.